### PR TITLE
PP-1254: PBS starts in the background even when started using init script much after the system startup

### PR DESCRIPTION
**Pull request by nithinj**
#### Bug/feature Description
* *PBS starts in the background even when started using init script much after the system startup*
* *How to reproduce??*
1. PBS is not registered with systemd for auto start up during the boot-time.
2. Restart the machine.
3. Start pbs
It will start in the background.
* *If there is an issue ID, link to it: [PP-1254](https://pbspro.atlassian.net/browse/PP-1254)*

#### Cause / Analysis / Design
PBS check whether boot_time > prev_start_time then it starts in the background.
This assumption is wrong when pbs is not registered to start at system start up.
Then the first start-up of pbs will be in backround even if the admin starts it by running init script.

#### Solution Description
* Checking whether the system is registered using chkconfig and systemctl
* If not returning from the function with false.
* Updated two smoke tests.

#### Testing logs/output
Attached in the Jira ticket.

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
- [x] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [x] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: src/cmds/scripts/pbs_init.d.in [hunk 1]
```diff
@@ -783,26 +783,42 @@ site_mom_cleanup()
 	fi
 }
```
Comment (suresh-thelkar) [inline src/cmds/scripts/pbs_init.d.in]: Can you please add a header describing briefly about the function.

File: src/cmds/scripts/pbs_init.d.in [hunk 2]
```diff
@@ -783,26 +783,42 @@ site_mom_cleanup()
...
 {
-	PYTHON_EXE=${PBS_EXEC}/python/bin/python
-	if [ -z "${PYTHON_EXE}" -o ! -x "${PYTHON_EXE}" ] ; then
-		return 1
-	fi
-	
-	BOOTPYFILE="/var/tmp/pbs_bootcheck.py"
-	BOOTCHECKFILE="/var/tmp/pbs_boot_check"
+	if is_registered
+	then
```
Comment (suresh-thelkar) [inline src/cmds/scripts/pbs_init.d.in]: Please combine the above two into single line. Alternatively you can check for failure of the function is_registered first and if it fails return from here itself. This way code looks much cleaner.

File: test/tests/pbs_smoketest.py [hunk 3]
```diff
@@ -1517,11 +1517,11 @@ def test_validate_pbsnodes_attrs(self):
             'sharing', 'last_state_change_time']
         possible_dflt_attrs = {'license': 'l', 'last_used_time': 'arbitrary'}
         self.server.expect(NODE, {'state': 'free'},
-                           id=self.server.shortname, interval=3)
+                           id=self.mom.shortname, interval=3)
```
Comment (suresh-thelkar) [inline test/tests/pbs_smoketest.py]: Is this change intended for this PR?

Comment (nithinj) [reply test/tests/pbs_smoketest.py]: Without this change the test will fail if the mom is running in a separate machine/container. I've added this along with the PR to save time as it is a single line change.

**Comment by nithinj** [comment]
Hi @suresh-thelkar ,
I've addressed your comments. Please check.

File: test/tests/pbs_smoketest.py (hunk 3 above)
Comment (suresh-thelkar) [reply test/tests/pbs_smoketest.py]: Since this is PTL bug you could have done it in different PR for better tracking. Since you have already done I am fine with this code as is.

**Comment by nithinj** [comment]
Thanks @suresh-thelkar  for your sign-off.

@PBSPro/pbspro-maintainers ,
Please have a look at this PR.

**Comment by agrawalravi90** [comment]
Wow, this has been lying around for 2+ weeks, sorry about that, Changes look fine to me, please rebase for merge

**Comment by agrawalravi90** [comment]
After you rebase, please run the PTL tests that you modified and attach their logs

**Comment by nithinj** [comment]
@agrawalravi90 ,
Thanks for the sign-off. I've re-based the branch. Modified PTL tests are part of smoke tests and they will be running as part of travis CI.

**Comment by mike0042** [comment]
What is the status of this PR? If ready for merge, please rebase.

**Comment by nithinj** [comment]
@mike0042 ,
I've re-based this branch. Please take it further.
